kolena/workflow/_datatypes.py

```diff
@@ -138,11 +139,13 @@ def serialize_value(value: Any) -> Any:
             raise ValueError(f"unsupported value type: '{type(value).__name__}' (value: {value})")
 
         items = [(field.name, getattr(self, field.name)) for field in dataclasses.fields(type(self))]
-        field_names = {field.name for field in dataclasses.fields(type(self))}
+        field_names = [field.name for field in dataclasses.fields(type(self))]
+        field_set = set(field_names)
         if _allow_extra(type(self)):
             for key, val in vars(self).items():
-                if key not in field_names and not _double_under(key):
+                if key not in field_set and not _double_under(key):
                     items.append((key, val))
```

Comment:
Should we also append this key to `field_names`? As-is, extra values are missing from the dumped `_field_order`.

Reply:
thanks for catching, updated with test.